components/scroll-to-top-button.tsx

```
@@ -0,0 +1,165 @@
+        className={cn(
+          "relative rounded-full transition-all duration-500 ease-out",
+          "flex items-center justify-center",
+          "hover:scale-110 active:scale-95",
+          "animate-fadeInUp",
+          "group",
+          sizeClasses[size],
+          variants[variant],
+          className
+        )}
+        aria-label="Scroll to top"
+      >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)